cli/fmt_errors.rs

```diff
@@ -22,7 +22,19 @@ pub fn format_location(frame: &JsStackFrame) -> String {
   }
   let mut result = String::new();
   if let Some(file_name) = &frame.file_name {
-    result += &cyan(&file_name).to_string();
+    let mut fmt_file_name = file_name.clone();
+    if fmt_file_name.starts_with("data:") {
+      if let Some(data_pieces) = fmt_file_name.split_once(',') {
+        let data_length = data_pieces.1.len();
+        if let Some(data_start) = data_pieces.1.get(0..5) {
+          if let Some(data_end) = data_pieces.1.get(data_length - 5..) {
+            fmt_file_name =
+              format!("{},{}...{}", data_pieces.0, data_start, data_end)
+          }
+        }
+      }
+    }
+    result += &cyan(fmt_file_name).to_string();
```

Comment:
This is very brittle - please use the data URL parser for this. Not all data URLs are contain base64 data. They can also contain plain text.

Reply:
Updated Rust and Javascript versions to use URL parsers. Modified the tests to demonstrate that it works the same regardless of data encoding.

Reply:
I agree with Luca that this bit should use data URL parser. We already have dependency on `data_url` crate, here's an example how it's used in other parts of the codebase: https://github.com/denoland/deno/blob/e02403bb4590da6823eb956bcbc89e9a937d431d/cli/file_fetcher.rs#L330-L336